tests/unit/test_actions.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from pathlib import Path
+from subprocess import CompletedProcess
+from unittest.mock import patch
+
+import yaml
+from ops import testing
+from ops.testing import ActionFailed
+from pytest import raises
+
+from charm import EtcdOperatorCharm
+from literals import INTERNAL_USER, PEER_RELATION
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+def test_get_password():
+    ctx = testing.Context(EtcdOperatorCharm)
+    relation = testing.PeerRelation(id=1, endpoint=PEER_RELATION)
+
+    # make sure admin credentials are created initially
+    state_in = testing.State(relations={relation}, leader=True)
+    state_out = ctx.run(ctx.on.leader_elected(), state_in)
+
+    ctx.run(ctx.on.action("get-password", params={"username": f"{INTERNAL_USER}"}), state_out)
+    assert ctx.action_results.get("username") == INTERNAL_USER
```

Comment:
nit: `assert ctx.action_results`

Reply:
This will not work as we get back the whole dict.

Reply:
This is nitpicking to silence pylance. I didn't suggest to make that the whole test just to add the test to make sure the 'action_results' is not None